FitnessClub/Data/BLL/Services/CoachService.cs

```diff
@@ -0,0 +1,135 @@
+﻿using FitnessClub.Data.BLL.Interfaces;
+using FitnessClub.Data.DAL.Interfaces;
+using FitnessClub.Data.DAL.Utility;
+using FitnessClub.Data.Models.ViewModels;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace FitnessClub.Data.BLL.Services
+{
+    public class CoachService : EmployeeService, ICoachService
+    {
+        private readonly ISessionRepository sessionRepository;
+        private readonly ICoachRepository coachRepository;
+        private readonly int userId;
+        public CoachService(
+            ISessionRepository sessionRepository,
+            IEmployeeRepository employeeRepository,
+            ICoachRepository coachRepository,
+            UserResolverService userResolverService
+            ) : base(sessionRepository, employeeRepository, userResolverService)
+        {
+            this.sessionRepository = sessionRepository;
+            this.coachRepository = coachRepository;
+            userId = userResolverService.GetUserId();
+        }
+
+        public async Task AssignToSession(int coachId, int sessionId)
+        {
+            var coach = await coachRepository.GetById(coachId);
+            var session = sessionRepository.FindWithCoach(sessionId);
+            if (coach == null )
```

Comment:
Maybe this check should be done immediately after assigning value to coach variable, so that we don't call sessionRepository when coach is null.

Reply:
Good point, fixed.